log4j-api/src/main/java/org/apache/logging/log4j/util/PropertiesUtil.java

```diff
@@ -746,6 +758,22 @@ public String getStringProperty(final String key) {
             return result;
         }
 
+        private String sourceGetProperty(ContextAwarePropertySource source, String contextName, String key) {
+            try {
+                return source.getProperty(contextName, key);
+            } catch (Throwable ex) {
+                return null;
```

Comment:
Shouldn't we better dump `ex` using the `StatusLogger` or something else?

Reply:
I don't think so. The issue that was reported simply causes error messages to be logged. The application is shutting down so an error during shutdown is irritating. Changing that to a warning doesn't really help much.